salt/modules/boto_secgroup.py

```diff
@@ -139,6 +139,10 @@ def _get_group(conn, name=None, vpc_id=None, group_id=None, region=None):  # pyl
             for group in filtered_groups:
                 # a group in EC2-Classic will have vpc_id set to None
                 if group.vpc_id is None:
+                    log.debug('Found EC2-Classic group: {0}'.format(group))
+                    return group
+                else:
+                    log.debug('Found EC2-VPC group: {0}'.format(group))
                     return group
```

Comment:
Notice in line 131 that this logic only gets called if the vpc_id argument is None. Returning group here is really, really dangerous for people using both VPC and Classic.

If you're using a VPC, you need to specify your VPC id via the execution module params or via the state, which will pass it to the execution module.

The behavior as-is ensures you don't accidentally return VPC groups when you intended to retrieve classic groups.